### PR TITLE
ADEN-13851 adding uap support for mtc

### PR DESCRIPTION
**Pull request by adi-wesolowski-fandom**

### CODE REVIEW

File: src/core/models/ad-slot.ts [hunk 1]
```diff
@@ -276,6 +276,9 @@ export class AdSlot {
 	}
 
 	getSlotName(): string {
+		if (context.get('state.extenal.adengine') === 'mtc' && this.config.slotName.includes('_')) {
```
Comment (preludek) [inline src/core/models/ad-slot.ts:279]: external? 😈 
```suggestion
		if (context.get('state.external.adengine') === 'mtc' && this.config.slotName.includes('_')) {
```

File: src/core/services/slot-service.ts [hunk 2]
```diff
@@ -107,6 +107,10 @@ class SlotService {
 			}
 		});
 
+		if (slotByPos === null && context.get('state.extenal.adengine') === 'mtc' && id.includes('-')) {
```
Comment (preludek) [inline src/core/services/slot-service.ts:110]:
```suggestion
		if (slotByPos === null && context.get('state.external.adengine') === 'mtc' && id.includes('-')) {
```

File: src/platforms/news-and-ratings/tvguide-mtc/setup/wiki-context.setup.ts [hunk 3]
```diff
@@ -1,7 +1,12 @@
-import { DiProcess, targetingService } from '@wikia/ad-engine';
+import { context, DiProcess, registerCustomAdLoader, targetingService } from '@wikia/ad-engine';
 
 export class TvGuideMTCContextSetup implements DiProcess {
 	execute(): void {
 		targetingService.set('skin', 'tvguide-mtc');
+
+		context.set('state.provider', 'nothing');
+		context.set('state.extenal.adengine', 'mtc');
```
Comment (preludek) [inline src/platforms/news-and-ratings/tvguide-mtc/setup/wiki-context.setup.ts:8]:
```suggestion
		context.set('state.external.adengine', 'mtc');
```

File: src/platforms/news-and-ratings/tvguide-mtc/templates/tvguide-mtc-templates.setup.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { DiProcess, logTemplates, TemplateRegistry, templateService } from '@wikia/ad-engine';
+import { Injectable } from '@wikia/dependency-injection';
+import { merge } from 'rxjs';
+import { registerBfaaTemplate } from '../../shared';
+// eslint-disable-next-line no-restricted-imports
```
Comment (preludek) [inline src/platforms/news-and-ratings/tvguide-mtc/templates/tvguide-mtc-templates.setup.ts:5]: T___T

**Review by preludek — APPROVED** [review]
Typos, overall looks good 👍

File: src/core/models/ad-slot.ts (hunk 1 above)
Comment (adi-wesolowski-fandom) [reply src/core/models/ad-slot.ts:279]: yup, should be external :)

File: src/platforms/news-and-ratings/tvguide-mtc/templates/tvguide-mtc-templates.setup.ts (hunk 4 above)
Comment (adi-wesolowski-fandom) [reply src/platforms/news-and-ratings/tvguide-mtc/templates/tvguide-mtc-templates.setup.ts:5]: its this or duplicating code... linter are very restricted (too much IMO) for paths of too many levels up

**Review by preludek — APPROVED** [review]

**Review by lnawrotek — APPROVED** [review]

**Pull request merged**